Adding favorites to user.

RominaMusacchio opened the pull request:
Adding option so add and remove user on the same endpoint (toggleFavorite)

Adding endpoint to list all favorites ids for an user.

strikerout commented on src/main/java/com/capitravel/Capitravel/controller/UserController.java:
```diff
@@ -43,4 +44,16 @@ public ResponseEntity<Void> addRoleToUser(@PathVariable String email, @RequestPa
     }
+
+    @PostMapping("/favorites")
+    public ResponseEntity<Set<Long>> toggleFavorite(@RequestParam String email, @RequestParam Long experienceId) {
```
hagamos a email un  PathParam, porque para emails que lleven el email+1@test.com
los +1,+2,+3 etc no funciona el requestParam

strikerout commented on src/main/java/com/capitravel/Capitravel/controller/UserController.java:
```diff
@@ -43,4 +44,16 @@ public ResponseEntity<Void> addRoleToUser(@PathVariable String email, @RequestPa
+    }
+
+    @GetMapping("/favorites")
+    public List<Long> listFavorites(@RequestParam String email) {
```
hagamos a email un  PathParam, porque para emails que lleven el email+1@test.com
los +1,+2,+3 etc no funciona el requestParam

RominaMusacchio committed:
Adding favorites to user.

Adding option so add and remove user on the same endpoint (toggleFavorite)

Adding endpoint to list all favorites ids for an user.

strikerout commented on src/main/java/com/capitravel/Capitravel/dto/UserDTO.java:
```diff
@@ -30,4 +32,6 @@ public class UserDTO {
     )
     private String password;
+
+    private List<Long> favoriteExperienceIds;
```
precisamos esto en el DTO?

strikerout commented on src/main/java/com/capitravel/Capitravel/dto/UserDTO.java:
```diff
@@ -6,6 +6,8 @@
 import jakarta.validation.constraints.Size;
 import lombok.Data;
 
+import java.util.Set;
```
no se usa este import

RominaMusacchio committed:
Fixing endpoints.
Removing favorites list from DTO

RominaMusacchio committed:
Removing import
